### ISSUE TITLE
deleteRow always logged value of the next  item which is still showing in the UI

### ISSUE THREAD

**Issue by vishnu947** [json-editor/json-editor#1241]
### General information

- json-editor version:
-2.9.0-beta.0

### Expected behavior
deleted data to be logged in when the deletedRow event got fired

### Actual behavior
deleteRow always get value of the next item as a deleted(while we check on console.log)  which is still in the UI

### Steps to reproduce the behavior
1. Any array if we want to delete and try to get the deleted Item through the deleteRow event 
2. Attatch deleteRow event 
3. add 4-5 rows and try to delete middle of them 
4. deleted row always logging the next item which is still in the UI
{
  "title": "Person",
  "type": "object",
  "required": [
    "name",
    "age",
    "date",
    "favorite_color",
    "gender",
    "location",
    "pets"
  ],
  "properties": {
    "name": {
      "type": "string",
      "description": "First and Last name",
      "minLength": 4,
      "default": "Jeremy Dorn"
    },
    "age": {
      "type": "integer",
      "default": 25,
      "minimum": 18,
      "maximum": 99
    },
    "favorite_color": {
      "type": "string",
      "format": "color",
      "title": "favorite color",
      "default": "#ffa500"
    },
    "gender": {
      "type": "string",
      "enum": [
        "male",
        "female",
        "other"
      ]
    },
    "date": {
      "type": "string",
      "format": "date",
      "options": {
        "flatpickr": {}
      }
    },
    "location": {
      "type": "object",
      "title": "Location",
      "properties": {
        "city": {
          "type": "string",
          "default": "San Francisco"
        },
        "state": {
          "type": "string",
          "default": "CA"
        },
        "citystate": {
          "type": "string",
          "description": "This is generated automatically from the previous two fields",
          "template": "{{city}}, {{state}}",
          "watch": {
            "city": "location.city",
            "state": "location.state"
          }
        }
      }
    },
    "pets": {
      "type": "array",
      "format": "table",
      "title": "Pets",
      "uniqueItems": true,
      "items": {
        "type": "object",
        "title": "Pet",
        "properties": {
          "type": {
            "type": "string",
            "enum": [
              "cat",
              "dog",
              "bird",
              "reptile",
              "other"
            ],
            "default": "dog"
          },
          "name": {
            "type": "string"
          }
        }
      },
      "default": [
        {
          "type": "dog",
          "name": "Walter"
        }
      ]
    }
  }
}
Direct link to example: https://json-editor.github.io/json-editor/

```javascript
if (jseditor instanceof window.JSONEditor) jseditor.destroy();
jseditor = new window.JSONEditor(document.querySelector("#json-editor-form"), jedata);
jseditor.on('deleteRow', editor => {
  console.log('deleteRow', editor.getValue())
});
```

**Comment by germanbisurgi** [comment]
When clicking "delete last item", the deleted **value** is passed to the  'deleteRow' event callback.

When deleting single items, the **instance** of the deleted editor is being passed to the 'deleteRow' event callback. This is  really not possible because the instance is already destroyed. The editor instance that its taking it place is returned instead.
I don't think this behaviour is useful. It would make more sense to return the deleted value here too. 

I also noticed that addRow is not returning the editor when format table is being used.

I would like to make (asap) a PR for this:

- fix addRow
- deleteLast and deleteItem returns value and not editor
- updape readme

**Comment by germanbisurgi** [comment]
Fixed in #1249
